File: 56-merge-intervals/56-merge-intervals.java
class Solution {
//     class com implements Comparator<int[]> {
//     @Override
//     public int compare(int[] a, int[] b) {
//         return a[0]-b[0];
//     }
// }
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals,(a,b) -> a[0] - b[0]);
        ArrayList<int[]> br=new ArrayList<>();
        int s=intervals[0][0];
        int e=intervals[0][1];
        int i=0,j=0;
        while(j<intervals.length){
            if(e>=intervals[j][0]){
                e=Math.max(e,intervals[j][1]);
                j++;
            }
            else if(e<intervals[j][0]){
                br.add(new int[]{intervals[i][0],e});
                i=j;
                if(i<intervals.length)
                    e=intervals[i][1];
            }
        }
        br.add(new int[]{intervals[i][0],e});
        return br.toArray(new int[br.size()][2]);
    }
}

File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    static bool compareInterval(vector<int> v1, vector<int> v2){
        return v1[0]<v2[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> res;
        
        sort(intervals.begin(),intervals.end(),compareInterval);
        auto itr=intervals.begin();
        int s=intervals[0][0],e=intervals[0][1];
        int i=0,j=1;
        while(j<intervals.size()){
            if(e>=intervals[j][0]){
                
                e=max(intervals[j][1],e);
                j++;
            }
            else{
                vector<int> v;
                v.push_back(intervals[i][0]);
                v.push_back(e);
                res.push_back(v);
                i=j;
                if(i<intervals.size()){
                    e=intervals[i][1];
                }
            }
        }
        vector<int> v1;
        v1.push_back(intervals[i][0]);
        v1.push_back(e);
        res.push_back(v1);
        return res;
    }
};